contracts/DxMgnPool.sol

```diff
@@ -200,6 +207,20 @@ contract DxMgnPool is Ownable {
         }
     }
 
+    function getAllClaimableMgnAndDeposits(address userAddress) external view returns(uint[] memory, uint[] memory) {
+        uint length = participationsByAddress[userAddress].length;
+
+        uint[] memory allUserClaimableMgn = new uint[](length);
+        uint[] memory allUserClaimableDeposit = new uint[](length);
+
+        for (uint i = 0; i < length; i++) {
+            allUserClaimableMgn[i] = calculateClaimableMgn(participationsByAddress[userAddress][i]);
+            allUserClaimableDeposit[i] = calculateClaimableDeposit(participationsByAddress[userAddress][i]);
```

Comment:
This will show a non-zero number even if the deposits have already been withdrawn.

Reply:
Can you elaborate?

Reply:
This doesn’t check it the deposit has already been withdrawn. We don’t delete the withdrawable deposit until the MGN is withdrawn as well. 

Let’s say you can withdraw 1EtH. After withdrawing the deposit this method would still return 1ETH as your “claimable deposit”

Reply:
Uh oh! This is no good. Would this fall in line with you suggestion to delete in the other PR?

Reply:
Not strictly related. The other one is just a small gas saving suggestion.

Reply:
good catch, we need to fix this